services/api/tests/src/test_admin_new_intersection.py

```diff
@@ -0,0 +1,113 @@
+from unittest.mock import patch, MagicMock, call
+import pytest
+import api.src.admin_new_intersection as admin_new_intersection
+import api.tests.data.admin_new_intersection_data as admin_new_intersection_data
+import sqlalchemy
+from werkzeug.exceptions import HTTPException
+
+###################################### Testing Requests ##########################################
+
+
+def test_request_options():
+    info = admin_new_intersection.AdminNewIntersection()
+    (body, code, headers) = info.options()
+    assert body == ""
+    assert code == 204
+    assert headers["Access-Control-Allow-Methods"] == "GET,POST"
+
+
+@patch("api.src.admin_new_intersection.get_allowed_selections")
+def test_entry_get(mock_get_allowed_selections):
+    req = MagicMock()
+    req.environ = admin_new_intersection_data.request_params_good
+    mock_get_allowed_selections.return_value = {}
+    with patch("api.src.admin_new_intersection.request", req):
+        status = admin_new_intersection.AdminNewIntersection()
+        (body, code, headers) = status.get()
+
+        mock_get_allowed_selections.assert_called_once()
+        assert code == 200
+        assert headers["Access-Control-Allow-Origin"] == "test.com"
+        assert body == {}
+
+
+@patch("api.src.admin_new_intersection.add_intersection")
+def test_entry_post(mock_add_intersection):
+    req = MagicMock()
+    req.environ = admin_new_intersection_data.request_params_good
+    req.json = admin_new_intersection_data.request_json_good
+    mock_add_intersection.return_value = {}, 200
+    with patch("api.src.admin_new_intersection.request", req):
+        status = admin_new_intersection.AdminNewIntersection()
+        (body, code, headers) = status.post()
+
+        mock_add_intersection.assert_called_once()
+        assert code == 200
+        assert headers["Access-Control-Allow-Origin"] == "test.com"
+        assert body == {}
+
+
+def test_entry_post_schema():
```

Comment:
Are we testing a bad schema here? Should we name it to reflect?

Reply:
renamed